python/GafferSceneUI/HistoryPathDialogue.py

```diff
@@ -0,0 +1,193 @@
+#
+#  Copyright (c) 2022, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import Gaffer
+import GafferUI
+import GafferScene
+import GafferSceneUI
+
+class _OperationIconColumn( GafferUI.PathColumn ) :
+
+	def __init__( self, title, property ) :
+
+		GafferUI.PathColumn.__init__( self )
+
+		self.__title = title
+		self.__property = property
+
+	def cellData( self, path, canceller = None ) :
+
+		cellValue = path.property( self.__property )
+
+		data = self.CellData()
+
+		data.icon = {
+			Gaffer.TweakPlug.Mode.Replace : "replaceSmall.png",
+			Gaffer.TweakPlug.Mode.Add : "plusSmall.png",
+			Gaffer.TweakPlug.Mode.Subtract : "minusSmall.png",
+			Gaffer.TweakPlug.Mode.Multiply : "multiplySmall.png",
+			Gaffer.TweakPlug.Mode.Remove : "removeSmall.png",
+			Gaffer.TweakPlug.Mode.Create : "createSmall.png",
+			Gaffer.TweakPlug.Mode.Min : "lessThanSmall.png",
+			Gaffer.TweakPlug.Mode.Max : "greaterThanSmall.png",
+			Gaffer.TweakPlug.Mode.ListAppend : "listAppendSmall.png",
+			Gaffer.TweakPlug.Mode.ListPrepend : "listPrependSmall.png",
+			Gaffer.TweakPlug.Mode.ListRemove : "listRemoveSmall.png",
+		}.get( cellValue, "errorSmall.png" )
+
+		return data
+
+	def headerData( self, canceller = None ) :
+
+		return self.CellData( self.__title )
+
+class HistoryPathDialogue( GafferUI.Dialogue ) :
```

Comment:
Does this benefit from deriving from Dialogue rather than Window? If it's not modal and it doesn't have buttons at the bottom, then I don't think it needs to be a dialogue at all...

Reply:
Fixed in fc273a94cb8aca738f9a8b3e65c10921a02b974a

Reply:
Since it's not a dialogue any more, and the path is pretty much an implementation detail, and we want to keep it protected for now, let's just call this `_HistoryWindow`.

Reply:
That's fixed and squashed down to 24ee00f71e93af8e0ab6694d9d7907b5facb2c3a.